src/utils/parser_server.rb

```diff
@@ -16,32 +12,61 @@
 trap(:INT) { quit = true }
 trap(:TERM) { quit = true }
 
-server = TCPServer.new(ARGV.first || 22_021)
+sockfile = ARGV.first
+server = UNIXServer.new(sockfile)
+
+at_exit do
+  server.close
+  File.unlink(sockfile)
+end
+
+MAX_LEN = 10 * 1024 * 1024
+
+def read_message(socket)
+  message = +''
+
+  loop do
+    message << socket.readpartial(MAX_LEN)
+  rescue EOFError
+    break
+  end
+
+  message.force_encoding('UTF-8').split('|', 2)
+end
 
 loop do
   break if quit
 
   # Start up a new thread that will handle each successive connection.
   Thread.new(server.accept_nonblock) do |socket|
-    message = socket.readpartial(10 * 1024 * 1024)
-    parser, source = message.force_encoding('UTF-8').split('|', 2)
+    parser, source = read_message(socket)
 
     response =
       case parser
       when 'ruby'
+        require_relative '../ruby/parser'
         Prettier::Parser.parse(source)
       when 'rbs'
+        require_relative '../rbs/parser'
```

Comment:
I think instead of requiring these every time (I assume you're doing this because they might not have `rbs` or `haml` installed, let's instead just provide a shim.

So in the `rbs` parser, we could do something like:

```ruby
begin
  require 'rbs'
rescue LoadError
  class Prettier::RBSParser
    def self.parse(source)
      false
    end
  end

  return
end

... rest of the file here ...
```

Same for HAML.